File: src/main/java/jp/kentan/minecraft/neko_event/config/provider/PlayerConfigProvider.java
package jp.kentan.minecraft.neko_event.config.provider;

import jp.kentan.minecraft.neko_event.config.ConfigManager;
import jp.kentan.minecraft.neko_event.util.Log;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.configuration.file.YamlConfiguration;

import java.io.File;
import java.util.Map;
import java.util.UUID;

public class PlayerConfigProvider {
    private static String sFolderPath;

    public static void setup(File dataFolder){
        sFolderPath = dataFolder + File.separator + "players" + File.separator;

        ConfigManager.createIfNeed(new File(sFolderPath));
    }

    static Object get(UUID uuid, String path, Object def) {
        final File file = new File(sFolderPath + uuid + ".yml");

        try {
            ConfigManager.createIfNeed(file);

            FileConfiguration config = new YamlConfiguration();
            config.load(file);

            return config.get(path, def);
        } catch (Exception e) {
            e.printStackTrace();
            Log.error(e.getMessage());
            return def;
        }
    }

    static boolean save(UUID uuid, Map<String, Object> dataList) {
        final File file = new File(sFolderPath + uuid + ".yml");

        try {
            ConfigManager.createIfNeed(file);

            FileConfiguration config = new YamlConfiguration();
            config.load(file);

            dataList.forEach(config::set);

            config.save(file);
        } catch (Exception e) {
            e.printStackTrace();
            Log.error(e.getMessage());
            return false;
        }

        return true;
    }
}

File: src/main/java/jp/kentan/minecraft/nekoevent/config/provider/PlayerConfigProvider.kt
package jp.kentan.minecraft.nekoevent.config.provider

import jp.kentan.minecraft.nekoevent.util.Log
import org.bukkit.configuration.file.YamlConfiguration
import java.io.File
import java.util.*

class PlayerConfigProvider(dataFolder: File) : BaseConfigProvider(dataFolder, "players${File.separator}") {

    @Throws(Exception::class)
    fun getConfig(uuid: UUID): Pair<File, YamlConfiguration> {
        val file = File(configFile, "$uuid.yml")
        return file to YamlConfiguration.loadConfiguration(file)
    }

    fun get(uuid: UUID, path: String): Any? {
        try {
            val config = getConfig(uuid).second
            return config.get(path)
        } catch (e: Exception) {
            e.printStackTrace()
            Log.error(e)
        }

        return null
    }

    fun save(uuid: UUID, dataMap: Map<String, Any>): Boolean {
        try {
            val file = File(configFile, "$uuid.yml")
            if (!file.exists()) {
                file.createNewFile()
            }

            val config = YamlConfiguration.loadConfiguration(file)
            dataMap.forEach(config::set)

            config.save(file)
        } catch (e: Exception) {
            e.printStackTrace()
            Log.error(e)
            return false
        }

        return true
    }
}